internal/devbox/packages.go

```diff
@@ -404,6 +414,9 @@ func (d *Devbox) installPackages(ctx context.Context, mode installMode) error {
 		if err := d.PluginManager().CreateFilesForConfig(pluginConfig); err != nil {
 			return err
 		}
+		if err := d.PluginManager().UpdateLockfileVersion(pluginConfig); err != nil {
+			return err
+		}
```

Comment:
I don't know if this is the best place to add this, because we'll be updating the lockfile _before_ the packages are actually installed in the nix store. In the event of an installation error, then the lockfile would be (wrongfully) updated.

In `ensureStateIsUpToDate`, we:
- call `d.installPackages()`
- ...// other operations that may also error
- call  `d.updateLockfile`

Could or should we move this `pluginManager.updateLockfileVersion` to `d.updateLockfile()`?

Reply:
yeah that makes sense. I put it here because I wanted to keep the existing logic intact (i.e. `d.PluginManager().CreateFilesForConfig` was previously doing this).

I think putting at the end is good. Another question is if we should move `CreateFilesForConfig` to the end as well, but at least for this PR, I'm inclined to leave as is.